### PR TITLE
🔧 PLAT-326 Buildkite improvements

### PR DESCRIPTION
**Pull request by dmihalcik-virtru**
* Create separate [repo]-pr pipeline for running on PRs and before deployment
* Create separate [repo]-deploy pipeline to allow the same master commit to deploy to multiple environments
* Shift stuff around to support this.

### CODE REVIEW

File: .buildkite/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  - label: ':house_with_garden: Create block for deploy environment selection'
+    branches: 'feature/* hotfix/*'
+    command: 'buildkite-scripts/mars/pipeline-steps/get-and-select-develop-environments.sh | buildkite-agent pipeline upload'
+
+  - wait
+
+  - label: ':house_with_garden: Set Environment'
+    command: 'buildkite-scripts/mars/scripts/set-deploy-environment.sh'
+    branches: 'master release/* feature/* hotfix/*'
```
Comment (jrschumacher) [inline .buildkite/deploy.yml]: @dmihalcik-virtru with this are we going to be eliminating `develop` branch model per previous conversations?

File: .buildkite/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+steps:
+  - label: 'annotate'
+    command: |
+      cat << EOF | buildkite-agent annotate --style "info"
+        [Triggered by ${BUILDKITE_TRIGGERED_FROM_BUILD_PIPELINE_SLUG} Build #${BUILDKITE_TRIGGERED_FROM_BUILD_NUMBER}](/${BUILDKITE_ORGANIZATION_SLUG}/${BUILDKITE_TRIGGERED_FROM_BUILD_PIPELINE_SLUG}/builds/${BUILDKITE_TRIGGERED_FROM_BUILD_NUMBER}#${BUILDKITE_TRIGGERED_FROM_BUILD_ID})
+      EOF
+    if: '"${BUILDKITE_SOURCE}" == "trigger_job"'
+
+  - label: ':house_with_garden: Create block for deploy environment selection'
```
Comment (jrschumacher) [inline .buildkite/deploy.yml]: This makes sense for blocking deployments, but does this do anything or does it drive line48?

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -21,63 +21,49 @@ env:
...
-      VIRTRU_ENV: develop01
-      S3_PATH: protect-develop
+    trigger: protect-and-track-deploy
+    build:
+      message: '[develop] ${BUILDKITE_MESSAGE}'
+      commit: '${BUILDKITE_COMMIT}'
+      branch: '${BUILDKITE_BRANCH}'
+      env:
+        BLOCK_DEPLOY: 'build.branch =~ /^feature\//'
+        VIRTRU_ENV: develop01
```
Comment (jrschumacher) [inline .buildkite/pipeline.yml]: Should we allow devs to select the environment to deploy to?

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -21,63 +21,49 @@ env:
...
-      VIRTRU_ENV: production
-      S3_PATH: 'protect/'
+    trigger: protect-and-track-deploy
+    build:
+      message: '[production] ${BUILDKITE_MESSAGE}'
+      commit: '${BUILDKITE_COMMIT}'
+      branch: '${BUILDKITE_BRANCH}'
+      env:
+        VIRTRU_ENV: production
+        S3_PATH: 'protect/'
```
Comment (jrschumacher) [inline .buildkite/pipeline.yml]: Why the trailing slash?

**Review by jrschumacher — COMMENTED** [review]
@dmihalcik-virtru left some comments, but otherwise looks good

File: .buildkite/deploy.yml (hunk 1 above)
Comment (dmihalcik-virtru) [reply .buildkite/deploy.yml]: yeah it is already gone

File: .buildkite/deploy.yml (hunk 2 above)
Comment (dmihalcik-virtru) [reply .buildkite/deploy.yml]: this sets the `deploy-environment-name` meta-data object, which is used to fill the screaming snake cased env variable later in the pipeline

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (dmihalcik-virtru) [reply .buildkite/pipeline.yml]: yeah probably but I'll do that later so not to block the rollout. Filed https://virtru.atlassian.net/browse/PLAT-329

File: .buildkite/pipeline.yml (hunk 4 above)
Comment (dmihalcik-virtru) [reply .buildkite/pipeline.yml]: No reason as far as I know.

File: .buildkite/deploy.yml (hunk 2 above)
Comment (jrschumacher) [reply .buildkite/deploy.yml]: 🐍🐍🐍

**Review by jrschumacher — DISMISSED** [review]

**Comment by dmihalcik-virtru** [comment]
@jrschumacher need a new review since I have strict reviewing enabled and pushed a rebase :-/

**Review by jrschumacher — APPROVED** [review]
LGTM

**Pull request merged**